subxt/src/utils/wrapper_opaque.rs

```diff
@@ -74,57 +74,45 @@ impl<T> WrapperKeepOpaque<T> {
 }
 
 impl<T> EncodeAsType for WrapperKeepOpaque<T> {
-    fn encode_as_type_to(
+    fn encode_as_type_to<R: TypeResolver>(
         &self,
-        type_id: u32,
-        types: &scale_info::PortableRegistry,
+        type_id: &R::TypeId,
+        types: &R,
         out: &mut Vec<u8>,
     ) -> Result<(), scale_encode::Error> {
         use scale_encode::error::{Error, ErrorKind, Kind};
 
-        let Some(ty) = types.resolve(type_id) else {
-            return Err(Error::new(ErrorKind::TypeNotFound(type_id)));
-        };
-
-        // Do a basic check that the target shape lines up.
-        let scale_info::TypeDef::Composite(_) = &ty.type_def else {
-            return Err(Error::new(ErrorKind::WrongShape {
+        let visitor = visitor::new(out, |_, _| {
+            // Check that the target shape lines up: any other shape but composite it wrong.
+            Err(Error::new(ErrorKind::WrongShape {
                 actual: Kind::Struct,
-                expected: type_id,
-            }));
-        };
-
-        // Check that the name also lines up.
-        if ty.path.ident().as_deref() != Some("WrapperKeepOpaque") {
```

Comment:
Dq: We had this check here before, but with the new type resolvers, it does not seem possible anymore with the new resolving strategy. What should we do here?

Reply:
I can see a couple of small arguments for exposing the enum name and type name (perhaps optionally only) in `scale-type-resolver`. For now, I think the point of this is that the type has slightly different encoding, and so we don't want to use it when the actual input wasn't also using it.

I think we can remove the check for now, but let's add a TODO in the code and/or issue to add support for optional names in `scale-type-resolver` :)

Reply:
Side note: this is good feedback too; if you do find other details that are missing like this, then please do make note of them, perhaps in issues in `scale-type-resolver` pointing to where a thing was needed, and then we can consider adding some bits in one go!